tests/models/test_version.py

```diff
@@ -301,6 +301,28 @@ def test_invalid_version_specs(self):
             VersionSpec("~")
         with pytest.raises(InvalidVersionSpec):
             VersionSpec("^")
+        # Test for previously mishandled uses of local version separator '+'
+        # uncovered by fuzzing efforts
+        with pytest.raises(InvalidVersionSpec):
+            VersionSpec("1.2+")
+        with pytest.raises(InvalidVersionSpec):
+            VersionSpec("+1.2")
+        with pytest.raises(InvalidVersionSpec):
+            VersionSpec("+1.2+")
+        with pytest.raises(InvalidVersionSpec):
+            VersionSpec("++")
+        # Fuzzer-identified crasher:
+        with pytest.raises(InvalidVersionSpec):
+            VersionSpec("c +, 0/|0 *")
+
+        # Test for mishandling of '==' without a version number
+        with pytest.raises(InvalidVersionSpec):
+            VersionSpec("==")
```

Comment:
Should we add this test case?
```
with pytest.raises(InvalidVersionSpec):
    VersionSpec("=")
```

If `==` is invalid, then `=` (single equals) should be as well.

Reply:
That case was "working" before so I didn't change it.  Current behavior is:
 ```conda search "numpy="``` is equivalent to ```conda search "numpy"```

It could be a breaking change for some people (but I agree with your assessment of how it SHOULD work).

Reply:
Let's open a spike to investigate this further.

Reply:
See https://github.com/conda/conda/issues/12046